Support edit yaml for catalog without questions

loganhz opened the pull request:
https://github.com/rancher/rancher/issues/13064
https://github.com/rancher/rancher/issues/9771

![image](https://user-images.githubusercontent.com/21168270/39954712-28d00d8c-55f6-11e8-99bd-d44311769923.png)
![image](https://user-images.githubusercontent.com/21168270/39954715-3f03280a-55f6-11e8-996b-a73a4c30728e.png)

loganhz commented:
FYI @vincent99 conflits resolved

vincent99 commented:
This is kind of weird... it works with ones that do have questions too but doesn't convert bools correctly, it turns structures into dotted keys, the cancel button is really more of an edit as a form vs YAML and doesn't cancel anything, read from a file doesn't seem to do anything in either mode..

loganhz commented:
Hi @vincent99 ,

1.  it works with ones that do have questions too

     `input-custom-answers` is newly added in this PR for app without questions. The feature of  supporting edit yaml  for app with questions has been merged two weeks ago. 

2.  but doesn't convert bools correctly.

    the schema type of `answers` is `map[string]`.  So boolean doesn't look good in YAML format. But server only takes string.

3.  it turns structures into dotted keys

     the schema type of `answers` is `map[string]`. So it only takes dotted keys. Server doesn't support answers in json format or YAML format now.

4. the cancel button is really more of an edit as a form vs YAML and doesn't cancel anything

    Updated. Renamed it to `Edit as YAML` and `Edit as a Form`.

5. read from a file doesn't seem to do anything in either mode.

    It should be working. I have tested it with the following yaml.

```
metrics:
  enabled: false
  image: wrouesnel/postgres_exporter
  imageTag: v0.1.1
  imagePullPolicy: IfNotPresent
  resources:
    requests:
      memory: 256Mi
      cpu: 100m
```

deniseschannon commented:
https://github.com/rancher/rancher/issues/9772

westlywright commented:
@vincent99 are we good to move forward with this?

vincent99 commented:
@westlywright I haven't looked at it since changes were made, but yes if it works consistently

westlywright commented:
@loganhz I checked this out this morning to play with it and I saw a couple issues. If I use the `Edit as Yaml` on the `hl-couchdb` catalog app I get a few errors.

1. Using `tab` causes an uncaught error about tabs not being allowed. The driver-rke component uses the same yaml component as this and does not fire the same error when using the tab key. It looks like we handle the tab error on that page. 
2. While editing the file I get a lot of `unable to parse` uncaught exceptions. I'm not 100% sure why we dont get these on the rke page but again I'd prob investigate how that was implemented to understand why.
3. Rebase when you figure these out

westlywright commented on app/components/input-custom-answers/component.js:
```diff
@@ -0,0 +1,132 @@
+  _boundChange:  null,
+
+  didInsertElement() {
+    this.set('_boundChange', (event) => {
```
`this.set` => `set(this,`

westlywright commented on app/components/input-custom-answers/component.js:
```diff
@@ -0,0 +1,132 @@
+    },
+
+    showPaste() {
+      this.set('pasteOrUpload', true);
```
this.set => set(this,

westlywright commented on app/components/input-custom-answers/component.js:
```diff
@@ -0,0 +1,132 @@
+    },
+
+    cancel() {
+      this.set('pasteOrUpload', false);
```
this.set => set(this,

westlywright commented on app/components/input-custom-answers/component.js:
```diff
@@ -0,0 +1,132 @@
+    if ( isSafari ) {
+      return '';
+    } else {
+      return this.get('accept');
```
this.get => get(this,

loganhz commented:
@westlywright Updated! Thanks for your review
